### ISSUE TITLE
Chunks is not iterable when using source maps with ProfilingPlugin

### ISSUE THREAD

**Issue by 7rulnik** [webpack/webpack#6779]
**Do you want to request a *feature* or report a *bug*?**

bug

**What is the current behavior?**

Webpack doesn't work when we are using one of this devtools with ProfilingPlugin

- cheap-module-source-map
- cheap-source-map
- hidden-source-map
- inline-source-map
- nosources-source-map
- source-map

It throw error:

```js
TypeError: chunks is not iterable
    at compilation.hooks.afterOptimizeChunkAssets.tap (/Users/v7rulnik/projects/opensource/huy/node_modules/webpack/lib/SourceMapDevToolPlugin.js:108:26)
    at args (/Users/v7rulnik/projects/opensource/huy/node_modules/webpack/lib/debug/ProfilingPlugin.js:383:10)
    at SyncHook.eval (<anonymous>:9:1)
    at SyncHook.lazyCompileHook [as _call] (/Users/v7rulnik/projects/opensource/huy/node_modules/tapable/lib/Hook.js:35:21)
    at hooks.optimizeChunkAssets.callAsync.err (/Users/v7rulnik/projects/opensource/huy/node_modules/webpack/lib/Compilation.js:947:42)
    at AsyncSeriesHook.eval [as callAsync] (<anonymous>:8:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Users/v7rulnik/projects/opensource/huy/node_modules/tapable/lib/Hook.js:35:21)
    at hooks.additionalAssets.callAsync.err (/Users/v7rulnik/projects/opensource/huy/node_modules/webpack/lib/Compilation.js:943:36)
    at AsyncSeriesHook.eval [as callAsync] (<anonymous>:8:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Users/v7rulnik/projects/opensource/huy/node_modules/tapable/lib/Hook.js:35:21)
    at hooks.optimizeTree.callAsync.err (/Users/v7rulnik/projects/opensource/huy/node_modules/webpack/lib/Compilation.js:939:32)
    at AsyncSeriesHook.eval [as callAsync] (<anonymous>:8:1)
    at AsyncSeriesHook.lazyCompileHook [as _callAsync] (/Users/v7rulnik/projects/opensource/huy/node_modules/tapable/lib/Hook.js:35:21)
    at Compilation.seal (/Users/v7rulnik/projects/opensource/huy/node_modules/webpack/lib/Compilation.js:881:27)
    at hooks.make.callAsync.err (/Users/v7rulnik/projects/opensource/huy/node_modules/webpack/lib/Compiler.js:464:17)
    at _err0 (<anonymous>:13:1)
    at _addModuleChain (/Users/v7rulnik/projects/opensource/huy/node_modules/webpack/lib/Compilation.js:749:12)
    at processModuleDependencies.err (/Users/v7rulnik/projects/opensource/huy/node_modules/webpack/lib/Compilation.js:688:9)
    at process._tickCallback (internal/process/next_tick.js:150:11)
```

**If the current behavior is a bug, please provide the steps to reproduce.**

```js
var path = require("path");
const webpack = require('webpack')

module.exports = [
	"cheap-module-source-map",
	"cheap-source-map",
	"hidden-source-map",
	"inline-source-map",
	"nosources-source-map",
	"source-map"
].map(devtool => {
	console.log(devtool)
	return {
		mode: "development",
		entry: {
			bundle: "coffee-loader!./example.coffee"
		},
		output: {
			path: path.join(__dirname, "dist"),
			filename: `./[name]-${devtool}.js`
		},
		plugins: [
			new webpack.debug.ProfilingPlugin(),
		],
		devtool,
		optimization: {
			runtimeChunk: true
		}
	}
})
```

**What is the expected behavior?**
It should not throw error and it should generate bundle and file for profiler.

**Please mention other relevant information such as the browser version, Node.js version, webpack version, and Operating System.**

- webpack@4.1.1
- webpack-clI@2.0.12
- node@9.4.0
- macOS@10.12.6

**Comment by TheLarkInn** [comment]
Great catch!! Wouuld you be willing to submit a PR for this?

**Comment by byzyk** [comment]
Hi @TheLarkInn, I tried to fix this in #6788. Please check my comment there

**Comment by 7rulnik** [comment]
Seems that I am too late :D
